### ISSUE TITLE
Upgrade dependency "github.com/gin-gonic/gin"

### ISSUE THREAD

**Issue by Ben131-Go** [multiversx/mx-chain-go#4959]
## Background
Module `github.com/ElrondNetwork/elrond-go` depends on `github.com/gin-gonic/gin@v1.8.0`.<br>
<https://github.com/multiversx/mx-chain-go/blob/master/go.mod#L23><br>
However, comparing version `v1.8.0` of `github.com/gin-gonic/gin` from proxy.golang.org and github, there are inconsistencies.<br>
> [commit time of the copy on github.com](https://api.github.com/repos/gin-gonic/gin/commits/38eb5ac)
```txt
"committer": {
      "name": "GitHub",
      "email": "noreply@github.com",
      "date": "2022-05-30T07:16:10Z"
    }
```
> [commit time of the copy on proxy.golang.org](https://proxy.golang.org/github.com/gin-gonic/gin/@v/v1.8.0.info)
```txt
{"Version":"v1.8.0","Time":"2022-05-28T07:23:00Z"}
```
So the checksum from the code in github does not match the checksum saved in sum.golang.org. The `v1.8.0` tag of `github.com/gin-gonic/gin` might have been retagged after a minor edition on github. I guess you use proxy.golang.org to get dependencies, but that also shows that your project is depending on the copy of `github.com/gin-gonic/gin@v1.8.0` before its edition. Depending upon such inconsistent tag version may also result in some unexpected errors as well as build errors due to different proxy settings.<br>
For example, when someone who does not use proxy.golang.org, say `GOPROXY=direct`, attempts to get `github.com/gin-gonic/gin@v1.8.0`, the following error occurs.
```Go
go: downloading github.com/gin-gonic/gin v1.8.0
go: github.com/gin-gonic/gin@v1.8.0: verifying module: checksum mismatch
        downloaded: h1:gcAUpegZ91QWGm03w0CjjJ0+rjxmG3pbBbT0aiT82s8=
        sum.golang.org: h1:4WFH5yycBMA3za5Hnl425yd9ymdw1XPm4666oab+hv4=

SECURITY ERROR
This download does NOT match the one reported by the checksum server.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.

For more information, see 'go help module-auth'.
```
So, this is a reminder in the hope that you can get rid of this problematic version of project `github.com/gin-gonic/gin`.
### 1. Bump the version of dependency `github.com/gin-gonic/gin`
I would recommend bumping the version of `github.com/gin-gonic/gin` to a new release to ensure dependency copy in proxy.golang.org and github in sync.

## References
+ <https://proxy.golang.org/>

**Comment by iulianpascalau** [comment]
Thanks for reporting this. We won't fix this against `master` branch since it is already fixed in the upcoming `rc/v1.4.0` as seen here: https://github.com/multiversx/mx-chain-go/blob/21b061aeb173cf5f061eca3d0eddf8b7561bbce3/go.mod#L11
When we will finish testing rc/v1.4.0 and we are ready to deploy it on the mainnet we will finally merge the branch into the `master` branch. Will close this issue after we finalize the merge.

**Comment by Ben131-Go** [comment]
Thanks for your reply!

**Comment by iulianpascalau** [comment]
Fixed, as seen in the last release v1.4.8 & master branch:
https://github.com/multiversx/mx-chain-go/blob/bda270a6701f46a08e10a41f36d18ebc551fe483/go.mod#L10

Thanks for reporting this.
